### PR TITLE
Update performance tests

### LINKED ISSUE #257: Update performance test structure
### Problem Description

Performance tests can become more generalized if we specify the expected runtime per dataset, instead of per individual test cases.

### Expected behavior

The current performance tests are based on test cases that specify the transformer, the dataset generator and the sizes.

This should be changed to a more generic approach where each dataset generator indicates its expected output values for different data sizes, and the performance test simply takes a transformer and runs all the compatible datasets through it, validating against those expected output values.

### PR DESCRIPTION
**Pull request by katxiao**
Resolve #257

### CODE REVIEW

**Comment by csala** [comment]
@katxiao Should this be put as ready for review?

File: tests/performance/test_performance.py [hunk 1]
```diff
@@ -33,52 +31,126 @@ def get_fqn(obj):
...
+            A mapping of data type (str) to a list of transformers.
+    """
+    transformers = BaseTransformer.get_subclasses()
+    transformers_map = {}
+
+    for transformer in transformers:
+        input_type = transformer.get_input_type()
+        input_type_transformers = transformers_map.get(input_type, [])
+        input_type_transformers.append(transformer)
+        transformers_map[input_type] = input_type_transformers
```
Comment (amontanez24) [inline tests/performance/test_performance.py:58]: minor: you could make `transformers_map` a `default_dict`. So these lines can be changed to something like
```
transformers_map = defaultdict(list)
transformers_map[input_type].append(transformer)
```

Also , there will be a method exactly like this in the `transformers.__init__.py` called `get_transformers_by_type`

**Review by amontanez24 — APPROVED** [review]
One small comment, but besides that looks good!

**Comment by katxiao** [comment]
With this update, perf tests now take almost 2 hours, since we are now running more performance tests than before. Maybe we should reduce the number of dataset sizes that we run with.

File: tests/performance/test_performance.py (hunk 1 above)
Comment (katxiao) [reply tests/performance/test_performance.py:58]: Re-using the `get_transformers_by_type` method sounds good to me. I'll wait for that PR.

**Review by pvk-developer — APPROVED** [review]
LGTM!

Only comments I have are that this may fail with the future lint coming from #248 since there are missing the `get_performance_thresholds` description.

Maybe we can merge and fix later? 

This is the `PR` tha will make this fail in the future: https://github.com/sdv-dev/RDT/pull/276/files

**Pull request merged**